### PR TITLE
fix: issues related to current studio repo

### LINKED ISSUE #1255: Governance issues is not properly allined.
#### Describe the bug
Governance issues is not properly allined it feels different in the UI.

#### How to Reproduce
1. Go to asyncapi studio website.
2. Go to Studio settings section.
3. Go to Governance section.

- Screenshots
Image

### LINKED ISSUE #1278: Layout glitches when toggling navigation or editor panels
#### Describe the bug
Toggling the navigation or editor panels causes layout flickering and pane size jumps.

#### How to Reproduce

https://github.com/user-attachments/assets/de4158aa-4a9d-4871-8038-b747ee5a9f46

Steps to reproduce the issue. Attach all resources that can help us understand the issue:

(attached video)
- Navigation open, enable Editor: Layout flickers while the editor is opening
- Only Preview visible, enable Navigation: Pane sizes jump and the layout glitches
- Only Preview visiblem enable Editor: Flickering occurs during the transition

#### Expected behavior

https://github.com/user-attachments/assets/40e58c70-0763-4208-b36e-6b5b7201e568

#### Proposed Changes 
File : `Content.tsx`
- Wrap Content with `React.memo` to prevent unnecessary re-renders.
- Use `useMemo` for `secondPaneSize` and `secondPaneMaxSize`, returning 0 when both panels are disabled.
- Memoize `localStorageLeftPaneSize` and `localStorageRightPaneSize`.
- Extract pane resize handlers with `useCallback`.
- Memoize `navigationAndEditor` JSX and pane styles.
- Set inner `SplitPane` `primary="first"` for consistent behavior.

Happy to open a PR for this if the maintainers approve.

### LINKED ISSUE #1292: fix: mobile layout breaks with sidebar overlap and horizontal overflow in Studio
##  Bug description

On mobile devices, the Studio layout breaks due to improper responsive handling.

Multiple issues are observed:
- Sidebar overlaps main content
- Content is pushed under the sidebar
- Page overflows horizontally (scrolls to the right)
- UI appears stretched and congested

---

##  Steps to reproduce

1. Open https://studio.asyncapi.com on a mobile device (or DevTools mobile view)
2. Navigate to any section (e.g. Operations)
3. Open sidebar or interact with navigation
4. Observe layout issues

---

##  Current behavior

- Sidebar does not collapse properly
- Main content is partially hidden
- Horizontal scroll appears
- Layout breaks visually

---

##  Expected behavior

- Sidebar should behave as overlay or collapse properly
- Content should remain fully visible
- No horizontal overflow
- Layout should be responsive on mobile

---

##  Screenshots

Image

Image

---

##  Possible cause

- Missing responsive breakpoints
- Incorrect container width (likely `100vw` or fixed width)
- Sidebar not handled properly for mobile layout

---

##  Contribution

I can work on this issue and submit a PR.

### LINKED ISSUE #1326: feat(pulsar): improve channel with pulsar bindings
#### Reason/Context
Operation only render the channel address, whereas in case of Pulsar it is the combination of  :
* channel.bindings.pulsar.persistence
* server.bindings.pulsar.tenant
* channel.bindings.pulsar.namespace
* channel.address

#### Description
Could it be possible to reuse pulsar bindings to provide channel fullname ?

`<persistence>://<tenant>/<namespace>/<address>`

Thank you

### LINKED ISSUE #1330: Studio 3.1.0: Left panel shows inverted PUB/SUB labels for channels
Environment:

AsyncAPI Studio (Web): https://studio.asyncapi.com/

Spec version: 3.1.0

Description:
The left panel (channel list) shows all channels as PUB/SUB inverted compared to the actual operation actions defined in the document. The right panel (details) and CLI validation show correct directions.

Steps to reproduce:

Open Studio.

Load any valid AsyncAPI 3.1.0 document with channels that have both send and receive operations.

Observe the left panel: channels with receive operations are labeled as PUB, channels with send operations are labeled as SUB.

Click on a channel — the right panel shows the correct action (e.g., RECEIVE for a subscribed channel).

Expected behavior:
Left panel labels should match operation actions:

receive → SUB

send → PUB

Actual behavior:
Labels are inverted:

receive → PUB

send → SUB

Impact:
Misleading for new users. Causes confusion when validating contracts visually. Right panel is correct, CLI validation passes.

Attachments:

- minimal YAML reproducing the issue:
```yaml
asyncapi: '3.1.0'
id: 'urn:test'
info:
  title: Test
  version: '1.0.0'
servers:
  production:
    host: 'kafka:9092'
    protocol: kafka
channels:
  test-topic:
    address: test-topic
operations:
  test-receive:
    action: receive
    channel:
      $ref: '#/channels/test-topic'
```
- screenshot of left panel vs right panel.
Image

### LINKED ISSUE #1332: fix:  ssr: false not allowed with next/dynamic in Server Component
## Description
When running the Studio locally, the app fails to load with the 
following error:

## Error
ssr: false  is not allowed with  next/dynamic  in Server Components. 
Please move it into a Client Component.

## Location
File: apps/studio/src/app/page.tsx
Line 2:
const StudioWrapper = dynamic(() => import('@/components/StudioWrapper'), {ssr: false})

## Steps to Reproduce
1. Clone the repo
2. Run `pnpm install`
3. Run `pnpm dev` inside `apps/studio`
4. Open `localhost:3000`
5. See error in terminal and blank page in browser

## Expected Behavior
Studio should load successfully at localhost:3000

## Actual Behavior
Page fails with error:  ssr: false  is not allowed in Server Components

## Possible Fix
Add `'use client'` directive at the top of `apps/studio/src/app/page.tsx`

## Environment
- Next.js: 15.5.18
- OS: Windows 11 with WSL2 Ubuntu

### LINKED ISSUE #1333: [MICROGRANT] Set of issues for resolution under the Microgrant Program 2026-08
List of issues to resolve:
- https://github.com/asyncapi/studio/issues/1292

Design issue from Maya:
- https://github.com/asyncapi/studio/issues/1335

### LINKED ISSUE #1334: [MICROGRANT] Set of issues for resolution under the Microgrant Program 2026-08
List of issues:

- https://github.com/asyncapi/studio/issues/1330
- https://github.com/asyncapi/studio/issues/1278
- https://github.com/asyncapi/studio/issues/1326

### LINKED ISSUE #1335: Replace inconsistent icon libraries with AsyncAPI design system icons
Background

As part of the icon system audit completed in brand#119, Studio was identified as the primary area of concern. The audit found that Studio currently pulls icons from three different libraries, each with six different viewBox grid sizes, on a single page — none of which align with the AsyncAPI design system standard.

What needs to be done

- Replace inconsistent Studio icons with the correct icons from the AsyncAPI design system
- Use the icon mapping documented in brand#119 as the implementation guide; it shows exactly which icon to replace and what to replace it with

References

Full audit and icon mapping: brand#119

cc @Shurtu-gal @aeworxet

### PR DESCRIPTION
**Pull request by Shurtu-gal**
**Description**

- Restored Studio build compatibility by rolling Next.js back to v14.
- Fixed layout flickering and pane-size jumps when toggling Navigation and Editor panels.
- Kept governance-setting labels on a single line.
- Display Pulsar operation channel names as `<persistence>://<tenant>/<namespace>/<address>` when complete, unambiguous bindings are available; otherwise retain the channel address.
- Extracted protocol-aware channel display formatting into an extensible helper registry.

**Related issue(s)**

Fixes #1334, #1332, #1326, #1278, #1255, #1330
Fixes #1333, #1335, #1292

### CODE REVIEW

**Comment by Shurtu-gal** [comment]
@Mayaleeeee only web icon not replaced as icon could not be found in figma.
Rest could you review the website once?

File: apps/studio/package.json [hunk 1]
```diff
@@ -87,7 +88,7 @@
     "mermaid": "^11.16.1",
     "monaco-editor": "0.34.1",
     "monaco-yaml": "4.0.2",
-    "next": "15.5.21",
+    "next": "14.2.35",
```
Comment (princerajpoot20) [inline apps/studio/package.json:91]: umm... why do we have to downgrade the version? If we do so, the dependency bot is going to override it. And IMHO, being on an older version isn't ideal. We can restructure `page.tsx` if needed.

File: apps/studio/src/components/Content.tsx [hunk 2]
```diff
@@ -6,21 +6,38 @@ import { Template } from './Template';
...
-import { FunctionComponent } from 'react';
+import { FunctionComponent, useEffect, useLayoutEffect, useState } from 'react';
 
 interface ContentProps {}
 
 export const Content: FunctionComponent<ContentProps> = () => { // eslint-disable-line sonarjs/cognitive-complexity
   const { show, secondaryPanelType } = usePanelsState();
+  const [isMobile, setIsMobile] = useState(() => window.matchMedia('(max-width: 767px)').matches);
   const document = useDocumentsState(state => state.documents['asyncapi']?.document) || null;
   const isV3 = document?.version() === '3.0.0';
```
Comment (princerajpoot20) [inline apps/studio/src/components/Content.tsx]: #1330 says: on a 3.1.0 document, left panel Pub/Sub labels are inverted.

This exact check means 3.1.0 still uses navigation  (Pub/Sub) instead of Navigationv3 (Receive/Send).
can we treat any 3.x as v3 so 3.1.0 gets Navigationv3

File: apps/studio/src/components/Navigation.tsx [hunk 3]
```diff
@@ -91,35 +91,38 @@ const OperationsNavigation: React.FunctionComponent<NavigationSectionProps> = ({
...
+                `operation-publish-${channelAddress}`,
               );
             }}
           >
             <div className="flex flex-row">
               <div className="flex-none">
-                <span className="mr-3 text-xs uppercase text-blue-500 font-bold">
-                  Pub
+                <span className="mr-3 text-xs uppercase text-green-500 font-bold">
+                  Sub
```
Comment (princerajpoot20) [inline apps/studio/src/components/Navigation.tsx:125]: Related to above: label swap helps only while 3.x wrongly hits this file.

i would say, prefer fixing isV3 so 3.x uses Navigationv3

If this file stays for 2.x only, confirm 2.x preview ids still match.

for 3.x, rely on Navigationv3 + operation-receive / operation-send

File: apps/studio/src/helpers/channel-name.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  return `${persistence}://${tenants[0]}/${namespace}/${address}`;
+}
+
+const protocolChannelNameFormatters: Record<string, ChannelNameFormatter> = {
+  pulsar: formatPulsarChannelName,
+};
+
+export function getChannelDisplayName(channel: ChannelInterface): string {
```
Comment (princerajpoot20) [inline apps/studio/src/helpers/channel-name.ts:50]: can we add tests for happy path + fallbacks for this. edge cases (missing namespace, multiple tenants ->fallback)

File: apps/studio/src/helpers/channel-name.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  const address = channel.address();
```
Comment (princerajpoot20) [inline apps/studio/src/helpers/channel-name.ts:51]: aslo getChannelDisplayName is used in old Navigation, not in v3 channel list (channel.address() still).

default Studio docs are often 3.0.0 → Navigationv3. So Pulsar improvement may not show for the main path. so probably we can use the helper in Navigationv3 channel rows

File: packages/ui/components/icons/AlertIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const AlertIcon = (props: any) => (
```
Comment (princerajpoot20) [inline packages/ui/components/icons/AlertIcon.tsx]: can we preffer React.SVGProps<SVGSVGElement> instead of any

File: apps/studio/src/app/globals.css [hunk 7]
```diff
@@ -2,6 +2,12 @@
 @tailwind components;
 @tailwind utilities;
 
+html,
+body {
+  max-width: 100%;
+  overflow-x: hidden;
```
Comment (princerajpoot20) [inline apps/studio/src/app/globals.css]: umm..overflow-x: hidden . i guess it can hide a layout bug... can we prefer keeping the flex/min-w-0

**Review by princerajpoot20 — CHANGES_REQUESTED** [review]
@Shurtu-gal left few comments

File: apps/studio/package.json (hunk 1 above)
Comment (Shurtu-gal) [reply apps/studio/package.json:91]: @princerajpoot20 this is quick fix for now. Right now the whole release pipeline is broken. There is nothing in the new version from which we are benefiting right now. So I don't see any point keeping us in NextJS 15.

Regarding dependabot, will put it in ignore list.

File: apps/studio/src/app/globals.css (hunk 7 above)
Comment (Shurtu-gal) [reply apps/studio/src/app/globals.css]: I am not sure I follow u on this. Could you please elaborate.

File: apps/studio/package.json (hunk 1 above)
Comment (Shurtu-gal) [reply apps/studio/package.json:91]: Done

File: apps/studio/src/components/Navigation.tsx (hunk 3 above)
Comment (Shurtu-gal) [reply apps/studio/src/components/Navigation.tsx:125]: Yeah updated the rule.

File: apps/studio/src/helpers/channel-name.ts (hunk 4 above)
Comment (Shurtu-gal) [reply apps/studio/src/helpers/channel-name.ts:50]: Done

File: packages/ui/components/icons/AlertIcon.tsx (hunk 6 above)
Comment (Shurtu-gal) [reply packages/ui/components/icons/AlertIcon.tsx]: Done

File: apps/studio/src/app/globals.css (hunk 7 above)
Comment (Shurtu-gal) [reply apps/studio/src/app/globals.css]: I think I kind of understood that. Have removed the global styling rules for now.

**Review by princerajpoot20 — APPROVED** [review]

**Review by magicmatatjahu — APPROVED** [review]

**Comment by Shurtu-gal** [comment]
Merging this for now @aeworxet. Will keep the parent bounty issues and icon issue open for design review maybe 🤔

**Comment by aeworxet** [comment]
@asyncapi/microgrant_team
